services/aster-nix/src/net/iface/util.rs

```diff
@@ -46,30 +48,39 @@ impl BindPortConfig {
 }
 
 pub fn spawn_background_poll_thread(iface: Arc<dyn Iface>) {
-    // FIXME: use timer or wait_timeout when timer is enable.
     let task_fn = move || {
-        debug!("spawn background poll thread");
+        trace!("spawn background poll thread for {}", iface.name());
+        let wait_queue = iface.polling_wait_queue();
         loop {
-            let next_poll_time = if let Some(next_poll_time) = iface.next_poll_at_ms() {
-                next_poll_time
+            let next_poll_at_ms = if let Some(next_poll_at_ms) = iface.next_poll_at_ms() {
+                next_poll_at_ms
+            }
+            // Sleep for a long time, if `next_poll_at_ms` is `None`.
+            else if let Some(next_poll_at_ms) = wait_queue
+                .wait_until_or_timeout(|| iface.next_poll_at_ms(), &Duration::from_secs(30))
```

Comment:
Personally, I prefer `wait_until` for this case, and this helps us avoid the "random" magic number `Duration::from_secs(30)` and make our code a bit cleaner.

Maybe you have a reason to use `wait_until_or_timeout` and specify a long time instead?

Reply:
Yes, introducing a magic number is not a good choice. It have been refactored.